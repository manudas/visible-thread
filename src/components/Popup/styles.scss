.popup {
    &__container {
        position: fixed;
        top: 0;
        width: 100%;
        height: 100%;
        display: flex;
        align-items: center;
        justify-content: center;
        background-color: rgba(110, 105, 92, 0.7);
    }
    &__content {
        padding: 30px;
        border-radius: 5px;
    }
}