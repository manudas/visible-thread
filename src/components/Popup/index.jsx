import PropTypes from 'prop-types';
import './styles.scss';

const Popup = ({ data, closePopup, text }) => {
    return data || text
        ? (
            <div
                className="popup__container"
                data-testid="popup__container"
                onClick={() => {
                    closePopup();
                    return false
                }}
            >
                <div
                    className="popup__content bg-light"
                >
                    {
                        data
                            ? data.map(({
                                totalScans,
                                teamName,
                            }, index) => {
                                return <div
                                    key={`${teamName}-${index}`}
                                    onClick={($event) => {
                                        $event.stopPropagation();
                                    }}
                                >
                                    Total scans in week {index}: {totalScans}
                                </div>
                            })
                        : text
                    }
                </div>
            </div>
        )
        : null;
}


Popup.propTypes = {
    closePopup: PropTypes.func,
    data: PropTypes.arrayOf(PropTypes.shape({
        totalScans: PropTypes.number,
        teamName: PropTypes.string
    })),
    text: PropTypes.string,
}

Popup.defaultProps = {
    closePopup: Function.prototype,
    data: null,
    text: null
}

export default Popup;
