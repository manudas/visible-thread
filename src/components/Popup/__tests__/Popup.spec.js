import React from 'react';
import {
    fireEvent,
    render,
} from '@testing-library/react';
import Popup from '..';

describe('Popup test suite', () => {

    it('should render <Popup /> with data', () => {
        let handler = jest.fn();
        let renderer = render(
            <Popup
                data={[
                    {
                        totalScans: 5,
                        teamName: "Team 1"
                    },
                    {
                        totalScans: 6,
                        teamName: "Team 2"
                    },
                ]}
                closePopup={handler}
            />
        );
        const { container } = renderer;
        expect(container).toMatchSnapshot();
    });

    it('should render <Popup /> with text', () => {
        let handler = jest.fn();
        let renderer = render(
            <Popup
                text="Test message"
                closePopup={handler}
            />
        );
        const { container } = renderer;
        expect(container).toMatchSnapshot();
    });

    it('should call close function', async () => {

        let handler = jest.fn();
        let renderer = render(
            <Popup
                text="Test message"
                closePopup={handler}
            />
        );

        const { getByTestId } = renderer;
        const container = getByTestId('popup__container');
        fireEvent.click(container);

        expect(handler).toHaveBeenCalledTimes(1);
    });
});
