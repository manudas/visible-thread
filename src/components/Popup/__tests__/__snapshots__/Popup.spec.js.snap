// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup test suite should render <Popup /> with data 1`] = `
<div>
  <div
    class="popup__container"
    data-testid="popup__container"
  >
    <div
      class="popup__content bg-light"
    >
      <div>
        Total scans in week 
        0
        : 
        5
      </div>
      <div>
        Total scans in week 
        1
        : 
        6
      </div>
    </div>
  </div>
</div>
`;

exports[`Popup test suite should render <Popup /> with text 1`] = `
<div>
  <div
    class="popup__container"
    data-testid="popup__container"
  >
    <div
      class="popup__content bg-light"
    >
      Test message
    </div>
  </div>
</div>
`;
